src/watchtower/actor.rs

```diff
@@ -131,7 +131,7 @@ where
                     }
                 }
             }
-            WatchtowerMessage::PeriodicCheck => self.periodic_check(state),
+            WatchtowerMessage::PeriodicCheck => self.periodic_check(state).await,
```

Comment:
Should `periodic_check` run in the background without blocking the actor to handle the next message?